### PR TITLE
video capture

### PR DESCRIPTION
**Pull request by kevodwyer**

### CODE REVIEW

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 1]
```diff
@@ -717,8 +717,11 @@ public static int getNumberOfChunks(long size) {
     }
 
     private CompletableFuture<Snapshot> recalculateThumbnail(Snapshot snapshot, Committer committer, String filename, AsyncReader fileData
-             , boolean isHidden, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
+             , boolean isHidden, long startIndex, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
     ) {
+        if(startIndex != 0) {
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java]: So in this case you're not recalculating the thumbnail? If so, then why not just avoid calling this method at all?

Comment (kevodwyer) [reply src/peergos/shared/user/fs/FileWrapper.java]: have refactored

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 2]
```diff
@@ -717,7 +725,7 @@ public static int getNumberOfChunks(long size) {
     }
 
     private CompletableFuture<Snapshot> recalculateThumbnail(Snapshot snapshot, Committer committer, String filename, AsyncReader fileData
-             , boolean isHidden, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
+             , boolean isHidden, long startIndex, long fileSize, NetworkAccess network, WritableAbsoluteCapability fileWriteCap, Optional<byte[]> streamSecret
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java]: no need for new parameter now

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 3]
```diff
@@ -654,24 +654,32 @@ public static int getNumberOfChunks(long size) {
                                                     throw new IllegalStateException("File already exists with name " + filename);
                                                 FileWrapper child = childOpt.get();
                                                 FileProperties childProps = child.getFileProperties();
+
+                                                TriFunction<FileWrapper, Snapshot, AsyncReader, CompletableFuture<Snapshot>> recalculateThumbnailIfNecessary =
+                                                        (updatedChild, latestSnapshot, is) -> {
+                                                    if(startIndex != 0) {
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java]: you could use an inline statement in both places. arguable whether or not that is simpler/shorter
e.g. startIndex != 0 ? Futures.of(latestSnapshot) : updatedChild.recalculateThumbnail(...)

**Pull request merged**